packages/@aws-cdk/cdk/test/core/test.token-json.ts

```diff
@@ -0,0 +1,175 @@
+
+export = {
+    'plain JSON.stringify() on a Token fails'(test: Test) {
+        // GIVEN
+        const token = new Token(() => 'value');
+
+        // WHEN
+        test.throws(() => {
+            JSON.stringify({ token });
+        });
+
+        test.done();
+    },
+
+    'string tokens can be JSONified and JSONification can be reversed'(test: Test) {
+        for (const token of tokensThatResolveTo('woof woof')) {
+            // GIVEN
+            const fido = { name: 'Fido', speaks: token };
+
+            // WHEN
+            const resolved = resolve(TokenJSON.stringify(fido));
+
+            // THEN
+            test.deepEqual(evaluateCFN(resolved), '{"name":"Fido","speaks":"woof woof"}');
+        }
+
+        test.done();
+    },
+
+    'string tokens can be embedded while being JSONified'(test: Test) {
+        for (const token of tokensThatResolveTo('woof woof')) {
+            // GIVEN
+            const fido = { name: 'Fido', speaks: `deep ${token}` };
+
+            // WHEN
+            const resolved = resolve(TokenJSON.stringify(fido));
+
+            // THEN
+            test.deepEqual(evaluateCFN(resolved), '{"name":"Fido","speaks":"deep woof woof"}');
+        }
+
+        test.done();
+    },
+
+    'integer Tokens behave correctly in stringification and JSONification'(test: Test) {
+        // GIVEN
+        const num = new Token(() => 1);
+        const embedded = `the number is ${num}`;
+
+        // WHEN
+        test.equal(evaluateCFN(resolve(embedded)), "the number is 1");
+        test.equal(evaluateCFN(resolve(TokenJSON.stringify({ embedded }))), "{\"embedded\":\"the number is 1\"}");
+        test.equal(evaluateCFN(resolve(TokenJSON.stringify({ num }))), "{\"num\":1}");
+
+        test.done();
+    },
+
+    'tokens in strings survive additional TokenJSON.stringification()'(test: Test) {
+        // GIVEN
+        for (const token of tokensThatResolveTo('pong!')) {
+            // WHEN
+            const stringified = TokenJSON.stringify(`ping? ${token}`);
+
+            // THEN
+            test.equal(evaluateCFN(resolve(stringified)), '"ping? pong!"');
+        }
+
+        test.done();
+    },
+
+    'intrinsic Tokens embed correctly in JSONification'(test: Test) {
+        // GIVEN
+        const bucketName = new CloudFormationIntrinsicToken({ Ref: 'MyBucket' });
+
+        // WHEN
+        const resolved = resolve(TokenJSON.stringify({ theBucket: bucketName }));
+
+        // THEN
+        const context = {MyBucket: 'TheName'};
+        test.equal(evaluateCFN(resolved, context), '{"theBucket":"TheName"}');
+
+        test.done();
+    },
+
+    'embedded string literals in intrinsics are escaped when calling TokenJSON.stringify()'(test: Test) {
+        // WHEN
+        const token = new FnConcat('Hello', 'This\nIs', 'Very "cool"');
+
+        // WHEN
+        const resolved = resolve(TokenJSON.stringify({
+            literal: 'I can also "contain" quotes',
+            token
```

Comment:
Can you please embed token into a string like this:

```
`I can also "contain" quotes: ${token}`
```

I'd like to make sure that this also works :-)

Reply:
https://github.com/awslabs/aws-cdk/pull/518/files/4206b85da3627ba586e8d773edbda076f2c41bf7#diff-9b9d373dbf84ff1ee826d2b7cdb757ffR135

and

https://github.com/awslabs/aws-cdk/pull/518/files/4206b85da3627ba586e8d773edbda076f2c41bf7#diff-9b9d373dbf84ff1ee826d2b7cdb757ffR151

both test variants of this